src/Counter.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+type Props = {};
+
+type State = {
+  count: number,
+};
+
+export class Counter extends React.Component<Props, State> {
+  state = {
+    count: 0,
+  };
+
+  addOne = () => {
+    this.setState((state) => ({
+      count: state.count + 1,
+    }));
+  };
+
+  add100 = () => {
+    this.setState((state) => ({
+      count: state.count + 100,
+    }));
+  };
+
+  increase = () => {
+    const currentCount = this.state.count;
+
+    if (currentCount % 5 === 0) {
+      this.add100();
+      this.addOne();
+    } else {
+      this.addOne();
+    }
```

Comment:
Don't repeat yourself
```suggestion
    if (currentCount % 5 === 0) {
      this.add100();
    }
    
   this.addOne();
```